### ISSUE TITLE
Allow Message to be set by file

### ISSUE THREAD

**Issue by arbourd** [arbourd/concourse-slack-alert-resource#36]

**Comment by chrishiestand** [comment]
It would be nice to be able to reference the file contents as a variable in `message` similar to the other slack resource.

**Comment by chrishiestand** [comment]
Even better (imo) would be to allow variables to be set from multiple files. The concourse yaml could look like this:

```yaml
- put: notify
  params:
    vars:
     - name: GIT_SHA1
       file_value: git-repo/.git/ref
     - name: VERSION
       file_value: git-repo/app-version.txt
    message: Successfully built app v$VERSION from commit https://github.com/arbourd/concourse-slack-alert-resource/commit/$GIT_SHA1
```

**Comment by avanier** [comment]
You could also go with a consolidated file of all the values. A single yaml of key -> value pairs for the stubs and their content.

This would require an upstream task to consolidate the values into a single file, but it might be more practical for certain applications. _(I feel it's the same amount of work but in different places, I just don't have my mind set as to either is actually a good idea >.< )_

```yaml
# mah_mighty.yml
pairs:
  VERSION: 1.2.3
  GIT_SHA1: 123123123

---
# pipeline.yml
- put: notify
  params:
    var_file: mah_mighty.yml
    message: Successfully built app v$VERSION from commit https://github.com/arbourd/concourse-slack-alert-resource/commit/$GIT_SHA1
```

**Comment by arbourd** [comment]
I think these are both two very good ideas, but I'm not sure if it's still better than having a small task in between that builds the message_file and passes it onto this task?

**Comment by chrishiestand** [comment]
I'm no longer on the project that was using this resource. I'll leave the thought process to the current active users.

**Comment by arbourd** [comment]
Closed with #47
